integration-tests/features/steps/common.py

```diff
@@ -22,14 +23,22 @@ def create_local_machines(context):
     even when *ensure_fresh* is set to 'no'. For faster test execution, relying
     on Ansible to restore the VM to a known state is recommended, rather than
     requiring a full VM create/destroy cycle.
+
+    When the tests are run as a non-root user, specifying 'as root' implies
+    'ensure_fresh=yes', even if it is otherwise set to 'no'
     """
+    as_root = (user == "root")
+    if user is not None and not as_root:
```

Comment:
I might misunderstand this condition but it seems to be wrong
As root can only be true when user is not none
So if user is not none it should be enough according to the error message 

Anyway I think that there is something mixed up here